### PR TITLE
portal: Spawn key derivation blocking

### PR DESCRIPTION
**Pull request by sophie-h**
This one depends on merging #21 and #22

### CODE REVIEW

File: src/portal/mod.rs [hunk 1]
```diff
@@ -283,24 +284,45 @@ impl Keyring {
...
+        let newkey = async_global_executor::spawn_blocking(move || {
+            async_std::task::block_on(async { keyring.read().await.derive_key(&secret) })
+        })
+        .await;
+
+        #[cfg(feature = "tokio")]
+        let newkey =
+            tokio::task::spawn_blocking(move || keyring.blocking_read().derive_key(&secret))
+                .await
+                .unwrap();
```
Comment (bilelmoussaoui) [inline src/portal/mod.rs:303]: is it safe to unwrap here?

Comment (sophie-h) [reply src/portal/mod.rs:303]: I mean it fails if the thread panics which would probably be on out of memory or whatever so I guess it's the best thing we can do?

Comment (bilelmoussaoui) [reply src/portal/mod.rs:303]: I would do an expect then, just to point out to the correct place to check in case of a potential future bug report

Comment (sophie-h) [reply src/portal/mod.rs:303]: I mean, unwraps always print the filename and line, right? Is there a reason this needs to be an expect and the other 56 unwraps not?

```
oo7 $ git g 'unwrap()' | wc -l
57
```

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**